### PR TITLE
Feat all cart and wishlist related functionalities

### PR DESCRIPTION
**Pull request by therajatg**
1. Can add item to cart from homepage.
2. Can directly go to cart if item already added via go to cart button.
3. Can add item to wishlist from homepage.
4. can remove item from wishlist from homepage.
5. Number of items in wishlist as well as in cart are shown in the navbar.

### CODE REVIEW

File: src/frontend/components/products/Products.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.post("/api/user/cart", {
+          headers: { authorization: encodedToken },
+        });
+        setCartItems(response.data.cart);
+      } catch (error) {
+        console.log(error);
+      }
```
Comment (raman1999) [inline src/frontend/components/products/Products.js:29]: remove console

File: src/frontend/components/products/Products.js [hunk 2]
```diff
@@ -1,41 +1,105 @@
...
+        },
+        {
+          headers: { authorization: token },
+        }
+      );
+      setWishlist(response.data.wishlist);
+    } catch (error) {
+      console.log(error);
+    }
+  };
```
Comment (raman1999) [inline src/frontend/components/products/Products.js:50]: remove console, you can display error message instead in UI

**Review by raman1999 — APPROVED** [review]
code looks clean, good breakdown of components

File: src/App.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ function App() {
         <Route path="/signup" element={<Signup />} />
         <Route path="/test-api" element={<Mockman />} />
       </Routes>
```
Comment (dikshadhote) [inline src/App.js:19]: Can make separate file for route and import it.

File: src/frontend/pages/login/login.js [hunk 4]
```diff
@@ -7,26 +7,39 @@ import axios from "axios";
...
-    } catch (error) {
-      authDispatch({ authType: "ERROR", authPayload: true });
-      console.log(`Error: ${error}`);
+    } catch (err) {
+      authDispatch({
+        type: "ERROR",
+        payload:
+          "Wrong Email or Pssword, signup if you don't have account with us",
+      });
+      console.log(err);
```
Comment (dikshadhote) [inline src/frontend/pages/login/login.js:31]: Really liked the idea of having different context for path and passing error messages as payload

**Review by dikshadhote — APPROVED** [review]
The code is  very neat and clean. Use of proper name and easy use of logic made it more readable. Good work !

File: src/App.js (hunk 3 above)
Comment (therajatg) [reply src/App.js:19]: Good suggestion. Done!

File: src/frontend/components/products/Products.js (hunk 2 above)
Comment (therajatg) [reply src/frontend/components/products/Products.js:50]: will do it after completing the ecom.

**Pull request merged**